### ISSUE TITLE
Getting the page of another domain instead of error 503 for https

### ISSUE THREAD

**Issue by grmat** [nginx-proxy/nginx-proxy#685]
Say I have the domain `example.org`, and running backends for `abc.example.org` and `test.example.org`. I have certs and everything set up nicely.
Now when I try to open `http://wiki.example.org`, I get a 503. That's alright.

But when I open `https://wiki.example.org`, I get served the page from the alphabetically first domain with a valid cert set up, in that case `abc.example.org`. No redirect, so obviously a `SSL_ERROR_BAD_CERT_DOMAIN` for `wiki.example.org`, but when I accept that, I see the `abc`-page.

IMHO this is unexpected and bad behaviour. Are there any suggestions I can prevent nginx from serving the wrong pages?

**Comment by bjoernbusch** [comment]
I just observed the same issue. Should there also be a default section in the template for 443, like for 80?

```
server {
	server_name _; # This is just an invalid value which will never trigger on a real hostname.
#	listen 80;
	listen 443;
	access_log /var/log/nginx/access.log vhost;
	return 503;
}
```

**Comment by bjoernbusch** [comment]
Ok, so the readme properly describes the issue and the template already contains exactly that section, but the section is only generated when there is a self-signed default certificate.

**Comment by bjoernbusch** [comment]
I've added two symbolic links `default.key` and `default.crt` to the root domain certificate, so that at least the config is properly generated and there is no forward to a "random" other domain. It's not perfect, but better than nothing.

**Comment by bjoernbusch** [comment]
#614

**Comment by nullsumme** [comment]
@bjoernbusch 
Hm. What if I don't have any root domain certificate? I have only the certs for the subdomains generated by letsencrypt.

**Comment by bjoernbusch** [comment]
I guess you could generate self signed certificates. I've never tried that manually myself and you would also see the undesired warning, but it would solve the wrong routing.

**Comment by nullsumme** [comment]
Yeah okay. But this sounds like a bad workaround :-|
Right now I think your first solution (default section for 443) looks the most promising.

**Comment by akamustang** [comment]
Just came to report the same issue.

**Comment by mehrad-rafigh** [comment]
Hi guys, I have the same issue here. Nginx is routing to other subdomains instead of returning an error code :/

**Comment by pawlakm** [comment]
5 years later the issue still exists: #2015

**Comment by buchdag** [comment]
This was fixed in #2186
